src/Button/elements.js

```diff
@@ -1,256 +1,61 @@
 import styled, { css } from 'styled-components';
-import { darken } from 'polished';
 
-/**
- * Gets the margin of the button.
- *
- * @param {String} margin - The given margin.
- * @param {Object} theme - The theme.
- *
- * @return {String}
- */
-const getMargin = (margin, theme) => {
-  if (!margin) {
-    return '0';
-  } else {
-    if (theme.dimensions[margin]) {
-      return theme.dimensions[margin];
-    } else {
-      return margin;
-    }
-  }
-};
-
-/**
- * Gets the radius for rounded buttons.
- *
- * @param {Number} padding - The padding.
- * @param {Object} theme - The theme.
- */
-
-export const ContainerIconRight = styled.span`
-  margin-left: 1rem;
-`;
-
-export const ContainerIconLeft = styled.span`
-  margin-right: 1rem;
-`;
+import { getAppearance, getSize } from './helpers';
 
+/* Button Container */
 export const Container = styled.button`
-    position: relative;
-
-    display: flex;
-    justify-content: center;
-    align-items: center;
+  position: relative;
 
-    margin-left: ${({ marginLeft, theme }) => getMargin(marginLeft, theme)};
-    margin-bottom: ${({ marginBottom, theme }) => getMargin(marginBottom, theme)};
-    padding: 0.8rem 1.6rem;
-    border-radius: ${({ theme: { dimensions } }) => dimensions.radius};
+  display: inline-flex;
+  justify-content: center;
+  align-items: center;
+  flex-wrap: nowrap;
 
-    font-size: ${({
-      theme: {
-        fonts: { size },
-      },
-    }) => size.medium};
-    line-height: ${({
-      theme: {
-        fonts: { size },
-      },
-    }) => size.medium};
-    color: ${({ theme: { palette } }) => palette.white};
+  max-width: 100%;
+  width: ${({ fluid }) => (fluid ? '100%' : 'auto')};
 
-    transition: background 200ms ease, box-shadow 200ms ease;
+  padding: 0.8rem 1.6rem;
```

Comment:
There are no variables in Theme/dimensions that could match here ? instead of hardcoding these ?

Reply:
@AntoninMarchardDev no